core/src/player.rs

```diff
@@ -110,6 +110,34 @@ impl StaticCallstack {
     }
 }
 
+#[derive(Collect)]
+#[collect(no_drop)]
+pub struct MouseData<'gc> {
+    /// The object that the mouse is currently hovering over.
+    hovered: Option<InteractiveObject<'gc>>,
+
+    /// If the mouse is down, the object that the mouse is currently pressing.
+    pressed: Option<InteractiveObject<'gc>>,
+}
+
+impl<'gc> MouseData<'gc> {
+    pub fn hovered_object(&self) -> Option<InteractiveObject<'gc>> {
```

Comment:
I think all these accessors are a bit overly verbose for how simple the struct is; on Discord I proposed making it a fully public struct, or going back to separate fields in UpdateContext (and just making them references)

Reply:
Made it a fully public struct, as otherwise the fields would be differentiated by their position in that huge tuple instead of their type in `update_context_params`.